### ISSUE TITLE
List of tuples initialized with a tuple adopts literal types

### ISSUE THREAD

**Issue by jparta** [microsoft/pyright#4976]
**Describe the bug**
When a list is initialized with a tuple with values within, the list adopts the type literally. Any subsequent tuples appended have to be the exact same literals.

**To Reproduce**
```
tuple_list = [("str", 0)]  # type is "list[tuple[Literal[''], Literal[0]]]"
tuple_list.append(("str2", 1))  # "Literal['str2']" cannot be assigned to "Literal['str']", "Literal[1]" cannot be assigned to "Literal[0]"
```
**Expected behavior**
I expect the list to take on a type of "list[tuple[str, int]]".

**VS Code extension or command-line**
extension v1.1.304

**Comment by erictraut** [comment]
This behavior is intended. Refer to [this documentation](https://microsoft.github.io/pyright/#/type-inference?id=tuple-expressions) about pyright's inference behaviors.

You can use a type annotation to influence the inference behaviors, as in the following:

```python
tuple_list: list[tuple[str, int]] = [("str", 0)]
```

**Comment by erictraut** [comment]
After much deliberation, I've decided to revisit this issue and change the way that pyright infers tuple expressions. In particular, when a tuple expression is embedded within another container (list, dict, set, or tuple) expression, the type checker will no longer retain literal values for the tuple. This change will be included in the next release of pyright.

**Comment by erictraut** [comment]
This is included in pyright 1.1.365.
